File: programming problems/HackerRank/30 days of code/day 2, operators/day_2_operators.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
    "strings"
)

/*
 * Complete the 'solve' function below.
 *
 * The function accepts following parameters:
 *  1. DOUBLE meal_cost
 *  2. INTEGER tip_percent
 *  3. INTEGER tax_percent
 */

func solve(meal_cost float64, tip_percent int32, tax_percent int32) {
    // Write your code here
    fmt.Printf("%.0f", meal_cost+meal_cost*float64(tip_percent)/100.0+meal_cost*float64(tax_percent)/100.0)
}

func main() {
    reader := bufio.NewReaderSize(os.Stdin, 16*1024*1024)

    meal_cost, err := strconv.ParseFloat(strings.TrimSpace(readLine(reader)), 64)
    checkError(err)

    tip_percentTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
    checkError(err)
    tip_percent := int32(tip_percentTemp)

    tax_percentTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
    checkError(err)
    tax_percent := int32(tax_percentTemp)

    solve(meal_cost, tip_percent, tax_percent)
}

func readLine(reader *bufio.Reader) string {
    str, _, err := reader.ReadLine()
    if err == io.EOF {
        return ""
    }

    return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
    if err != nil {
        panic(err)
    }
}

File: programming problems/HackerRank/30 days of code/day 2, operators/day_2_operators.rb
#!/bin/ruby

require 'json'
require 'stringio'

#
# Complete the 'solve' function below.
#
# The function accepts following parameters:
#  1. DOUBLE meal_cost
#  2. INTEGER tip_percent
#  3. INTEGER tax_percent
#

def solve(meal_cost, tip_percent, tax_percent)
    # Write your code here
    puts "%.0f" % (meal_cost + meal_cost*tip_percent/100 + meal_cost*tax_percent/100)
end

meal_cost = gets.strip.to_f

tip_percent = gets.strip.to_i

tax_percent = gets.strip.to_i

solve meal_cost, tip_percent, tax_percent
